src/ast.ts

```diff
@@ -239,19 +246,57 @@ export const replaceVariableDefinitions = (
 };
 
 /**
- * return the path that will be returned in the response from from the chain of parents
+ * Return the path that will be returned in the response from the chain of parents.
+ * By default this will only build up paths for field nodes, but the anyKind flag allows
+ * to build paths for any named node.
+ *
+ * It also supports aliases.
  */
 /** @hidden */
-export const extractPath = (parents: ReadonlyArray<ASTNode>): ReadonlyArray<string> => {
+export const extractPath = (
+  parents: ReadonlyArray<ASTNode>,
+  anyKind?: boolean
+): ReadonlyArray<string> => {
   const path: string[] = [];
-  parents.forEach(parent => {
-    if (parent.kind === 'Field') {
-      path.push(parent.name.value);
+  parents.forEach((parent: any) => {
+    if (anyKind) {
```

Comment:
nit: this condition and the subsequent conditional look like they could be combined into 1 with `if (parent.kind === 'Field' || anyKind) { ... }`

Reply:
So true!